services/7segmentdisplay.md

```diff
@@ -0,0 +1,33 @@
+#  7-segment display
+
+    identifier: 0x196158f7
+
+A classic 7-segment display, with 4 dots and column.
+
+## Registers
+
+    rw digits @ value {
+    repeats:
+        digit: u8
+        dot: bool
+    }
+
+The digit and dots displayed on the segment. If ``digit`` is outside ``0x0-0xf``, nothing is displayed.
```

Comment:
I think we should assign each bit to a position in the display. We can go A to bit 0, B to 1, ... G to 6, and dot to 7.

![150px-7_Segment_Display_with_Labeled_Segments svg](https://user-images.githubusercontent.com/10673976/105922679-6a701780-603b-11eb-8707-c557f111cadf.png)

This way you can encode the different ways of showing say the number 7, but you can also try to display some approximation of text. It would be up to client to keep a "font" - mapping from numbers (and possibly letters) to 8 bit patterns.

This also lets you clear the display. We should also say what happens if display has say 7 digits and you send 4 - I guess the last 3 should be set to 0.

Reply:
I changed to turn off the remaining leds?